lib/SILOptimizer/IPO/ClosureSpecializer.cpp

```diff
@@ -1098,23 +1100,49 @@ static int getSpecializationLevelRecursive(StringRef funcName, Demangler &parent
     return 0;
   if (funcSpec->getKind() != Node::Kind::FunctionSignatureSpecialization)
     return 0;
-  Node *param = funcSpec->getChild(1);
-  if (param->getKind() != Node::Kind::FunctionSignatureSpecializationParam)
-    return 0;
-  if (param->getNumChildren() < 2)
-    return 0;
-  Node *kindNd = param->getChild(0);
-  if (kindNd->getKind() != Node::Kind::FunctionSignatureSpecializationParamKind)
-    return 0;
-  auto kind = FunctionSigSpecializationParamKind(kindNd->getIndex());
-  if (kind != FunctionSigSpecializationParamKind::ConstantPropFunction)
-    return 0;
-    
-  Node *payload = param->getChild(1);
-  if (payload->getKind() != Node::Kind::FunctionSignatureSpecializationParamPayload)
-    return 1;
-  // Check if the specialized function is a specialization itself.
-  return 1 + getSpecializationLevelRecursive(payload->getText(), demangler);
+
+  // Handle up to six parameters, otherwise assume infinite recursion
+  if (funcSpec->getNumChildren() > 6)
```

Comment:
Do we need this limit to avoid exponential complexity?
Can you add a short comment, mentioning the reason for this limit?

Reply:
I don't think that anything can go wrong, even if the type tree is large. So, if the limit doesn't have any purpose, I suggest to remove it.

Reply:
Removed the argument limit